sktime/transformations/series/outlier_detection.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3 -u
+# -*- coding: utf-8 -*-
+
+__author__ = ["Martin Walter"]
+__all__ = ["HampelFilter"]
+
+from sktime.transformations.base import _SeriesToSeriesTransformer
+from sktime.utils.validation.series import check_series
+from sktime.forecasting.model_selection import SlidingWindowSplitter
+
+import numpy as np
+import warnings
+
+
+class HampelFilter(_SeriesToSeriesTransformer):
+    """HampelFilter to detect outliers based on a sliding window. Correction
+    of outliers is recommended by means of the sktime.Imputer,
+    so both can be tuned separately.
+
+    Parameters
+    ----------
+    window_length : int
+        Lenght of the sliding window
+    n_sigma : int, optional
+        Defines how strong a point must outly to be an "outlier", by default 3
+    k : float, optional
+        A constant scale factor which is dependent on the distribution,
+        for Gaussian it is approximately 1.4826, by default 1.4826
+    return_bool : bool, optional
+        If True, outliers are filled with True and non-outliers with False.
+        Else, outliers are filled with np.nan.
+
+    References
+    ----------
+    Hampel F. R., "The influence curve and its role in robust estimation",
+    Journal of the American Statistical Association, 69, 382–393, 1974
+
+    https://github.com/MichaelisTrofficus/hampel_filter
+    """
+
+    _tags = {"univariate-only": True, "fit-in-transform": True}
+
+    def __init__(self, window_length, n_sigma=3, k=1.4826, return_bool=False):
+
+        self.window_length = window_length
+        self.n_sigma = n_sigma
+        self.k = k
+        self.return_bool = return_bool
+        super(HampelFilter, self).__init__()
+
+    def transform(self, Z, X=None):
+        self.check_is_fitted()
+        z = check_series(Z, enforce_univariate=True)
+
+        # warn if nan values in Series, as user might mix them
+        # up with outliers otherwise
+        if z.isnull().values.any():
+            warnings.warn(
+                """Series contains nan values, more nan might be
+                added if there are outliers"""
+            )
+
+        cv = SlidingWindowSplitter(
+            window_length=self.window_length, step_length=1, start_with_window=True
+        )
+        half_window_length = int(self.window_length / 2)
+
+        for i in cv.split(z):
```

Comment:
for readability, I suggest to wrap this entire filter procedure line 68 - 98 into a `_hampel_filter(z, cv, half_window_length, ...)` function